app/admin/jobs/routine/[id]/client.tsx

```diff
@@ -0,0 +1,112 @@
+"use client";
+
+import React, { useState } from "react";
+import { testRoutineJobWorkflow } from "@/app/actions";
+
+interface RoutineJob {
+  id: string;
+  name: string;
+  cronExpression: string;
+  enabled: boolean;
+  createdAt: string;
+  sandboxUrl?: string | null;
+}
+
+interface RoutineJobDetailClientProps {
+  routineJob: RoutineJob;
+}
+
+export default function RoutineJobDetailClient({ routineJob }: RoutineJobDetailClientProps) {
+  const [testing, setTesting] = useState(false);
+  const [testResult, setTestResult] = useState<{ success: boolean; outputs?: unknown; message: string } | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const handleTest = async () => {
```

Comment:
Put all of test button component logic in its own client component, so that the rest of the page stays server side rendered